sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/ResourceEvaluationStage.java

```diff
@@ -63,40 +66,49 @@ public EvaluationOutcome evaluate(
             MesosResourcePool mesosResourcePool,
             OfferRequirement offerRequirement,
             OfferRecommendationSlate offerRecommendationSlate) {
-        ResourceRequirement resourceRequirement = getResourceRequirement();
-        Optional<MesosResource> mesosResourceOptional = mesosResourcePool.consume(resourceRequirement);
-        if (!mesosResourceOptional.isPresent()) {
-            return fail(this, "Failed to satisfy required resource '%s': %s",
-                    resourceRequirement.getName(),
-                    TextFormat.shortDebugString(resourceRequirement.getResource()));
-        }
+        final ResourceRequirement resourceRequirement = getResourceRequirement();
+        final String resourceId = resourceRequirement.getResourceId();
 
-        final MesosResource mesosResource = mesosResourceOptional.get();
-        Resource fulfilledResource = getFulfilledResource(mesosResource);
+        Resource fulfilledResource = getFulfilledResource(new MesosResource(resourceRequirement.getResource()));
         if (resourceRequirement.expectsResource()) {
             logger.info("Expects Resource");
 
+            Optional<MesosResource> existingResourceOptional = mesosResourcePool.getReservedResourceById(resourceId);
```

Comment:
It wouldn't work to fold this into the `consume` call below? I thought `MesosResourcePool` already handles this logic internally.